packages/datetime/src/dateRangeInput.tsx

```diff
@@ -286,10 +511,19 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         const { keys, values } = this.getStateKeysAndValuesForBoundary(boundary);
         const inputString = this.getFormattedDateString(values.selectedValue);
 
+        // change the boundary only if the user explicitly focused in the field.
+        // focus changes from hovering don't count; they're just temporary.
+        const preferredBoundaryToModify = (this.state.wasLastFocusChangeDueToHover)
+            ? this.state.preferredBoundaryToModify
+            : boundary;
+
         this.setState({
             isOpen: true,
+            preferredBoundaryToModify,
             [keys.inputString]: inputString,
             [keys.isInputFocused]: true,
+            mostRecentlyFocusedField: boundary,
```

Comment:
Gah, would really prefer to set this variable only in `handleInputFocus`. Don't remember off the top of my head why setting it here was necessary. Can take a look after this PR if that's okay.